unumpy/_multimethods.py

```diff
@@ -824,25 +804,442 @@ def nancumsum(a, axis=None, dtype=None, out=None):
     return (a, mark_dtype(dtype), mark_non_coercible(out))
 
 
-@create_numpy(_reduce_argreplacer)
+@create_numpy(_first2argreplacer)
 @all_of_type(ndarray)
-def std(a, axis=None, dtype=None, out=None, ddof=0, keepdims=False):
+def percentile(
+    a,
+    q,
+    axis=None,
+    out=None,
+    overwrite_input=False,
+    interpolation="linear",
+    keepdims=False,
+):
+    return (a, q, mark_non_coercible(out))
+
+
+@create_numpy(_first2argreplacer)
+@all_of_type(ndarray)
+def nanpercentile(
+    a,
+    q,
+    axis=None,
+    out=None,
+    overwrite_input=False,
+    interpolation="linear",
+    keepdims=False,
+):
+    return (a, q, mark_non_coercible(out))
+
+
+@create_numpy(_first2argreplacer)
+@all_of_type(ndarray)
+def quantile(
+    a,
+    q,
+    axis=None,
+    out=None,
+    overwrite_input=False,
+    interpolation="linear",
+    keepdims=False,
+):
+    return (a, q, mark_non_coercible(out))
+
+
+@create_numpy(_first2argreplacer)
+@all_of_type(ndarray)
+def nanquantile(
+    a,
+    q,
+    axis=None,
+    out=None,
+    overwrite_input=False,
+    interpolation="linear",
+    keepdims=False,
+):
+    return (a, q, mark_non_coercible(out))
+
+
+def _ureduce(a, axis):
+    nd = ndim(a)
+
+    if axis is None:
+        a = ravel(a)
+        dims = (1,) * nd
+    else:
+        if not isinstance(axis, collections.abc.Sequence):
+            axis = (axis,)
+
+        axis = _normalize_axis(nd, axis)
+        unselected_axis = tuple(set(range(nd)) - set(axis))
+
+        dims = list(a.shape)
+        for ax in axis:
+            dims[ax] = 1
+        dims = tuple(dims)
+
+        a = transpose(a, unselected_axis + axis)
+        a = a.reshape(a.shape[: len(unselected_axis)] + (-1,))
+
+    return a, dims
+
+
+def _median_default(a, axis=None, out=None, overwrite_input=False, keepdims=False):
+    if overwrite_input:
+        return NotImplemented
+
+    a, dims = _ureduce(a, axis)
+
+    mask = any(isnan(a), axis=-1).reshape(a.shape[0:-1] + (1,))
+    a = where(mask, nan, a)
+
+    a = sort(a, axis=-1)
```

Comment:
Any reason no to use [`partition`](https://numpy.org/doc/stable/reference/generated/numpy.partition.html?highlight=partition#numpy.partition)?

Reply:
I guess it's mostly for simplicity. Is `partition` more efficient?

Reply:
It should be slightly more efficient as it doesn't completely sort the array.

Reply:
I replaced `sort` with `partition` as you suggested.